receiver/prometheusexec/subprocessmanager/manager.go

```diff
@@ -0,0 +1,145 @@
+	initialDelay time.Duration = 1 * time.Second
+)
+
+// Run will start the process and keep track of running time
+func (proc *Process) Run(logger *zap.Logger) (time.Duration, error) {
+	var (
+		start     time.Time
+		elapsed   time.Duration
+		argsSlice []string
+	)
+
+	// Parse the command line string into arguments
+	args, err := shellquote.Split(proc.Command)
+	if err != nil {
+		return elapsed, fmt.Errorf("[%v] could not parse command error: %v", proc.CustomName, err)
+	}
+	// Separate the executable from the flags for the Command object
+	if len(args) > 1 {
+		argsSlice = args[1:]
+	}
+
+	// Create the command object and attach current os environment + environment variables defined by user
+	childProcess := exec.Command(args[0], argsSlice...)
+	childProcess.Env = os.Environ()
+	childProcess.Env = append(childProcess.Env, formatEnvSlice(&proc.Env)...)
+
+	// Handle the subprocess output in a goroutine, and pipe the stderr/stdout into one stream
+	cmdReader, err := childProcess.StdoutPipe()
+	if err != nil {
+		return elapsed, fmt.Errorf("[%v] could not get the command's output pipe, err: %v", proc.CustomName, err)
+	}
+	childProcess.Stderr = childProcess.Stdout
+	go proc.pipeSubprocessOutput(bufio.NewReader(cmdReader), logger)
+
+	// Start and stop timer (elapsed) right before and after executing the command
+	start = time.Now()
+	errProcess := childProcess.Start()
+	if errProcess != nil {
+		return elapsed, fmt.Errorf("[%v] process could not start: %v", proc.CustomName, errProcess)
+	}
+
+	errProcess = childProcess.Wait()
+	elapsed = time.Since(start)
+	if errProcess != nil {
+		return elapsed, fmt.Errorf("[%v] process error: %v", proc.CustomName, errProcess)
+	}
+
+	return elapsed, nil
+}
+
+// Log every line of the subprocesse's output using zap
+func (proc *Process) pipeSubprocessOutput(reader *bufio.Reader, logger *zap.Logger) {
+	// Infinite reading loop until EOF (pipe is closed)
+	for {
+		line, err := reader.ReadString('\n')
+		if err != nil && err != io.EOF {
+			logger.Info("subprocess logging failed", zap.String("subprocess name", proc.CustomName), zap.String("error", err.Error()))
+			break
+		}
+
+		line = strings.TrimSpace(line)
+		if line != "" {
+			logger.Info("subprocess output line", zap.String("subprocess name", proc.CustomName), zap.String("output", line))
+		}
+
+		// Leave this function when error is EOF (stderr/stdout pipe was closed)
+		if err == io.EOF {
+			break
+		}
+	}
+}
+
+// formatEnvSlice will loop over the key-value pairs and format the slice correctly for use by the Command object ("name=value")
+func formatEnvSlice(envs *[]config.EnvConfig) []string {
+	if len(*envs) == 0 {
+		return nil
+	}
+
+	envSlice := make([]string, len(*envs))
+	for i, env := range *envs {
+		envSlice[i] = fmt.Sprintf("%v=%v", env.Name, env.Value)
+	}
+
+	return envSlice
+}
+
+// GetDelay will compute the exponential backoff for a given process according to its crash count and time alive
```

Comment:
```suggestion
// GetDelay will compute the delay for a given process using an exponential backoff algorithm according to its crash count and time alive
```

Also consider if it would make sense to use https://github.com/cenkalti/backoff instead of writing this logic yourself. I think that package is used in a few other places in the Collector already

Reply:
Changed the wording! I prefer using my own logic since I have more power with the use of computing the backoff with `crashCount` as well as runtime (`elapsed`). Since I'm also managing the receiver associated to the subprocess, there's lots of different conditions that define whether a process is "healthy" or not!